### PR TITLE
feat: Add helm charts for containerised agw

### PR DESCRIPTION
**Pull request by ymasmoudi**
Signed-off-by: YOUSSEF EL MASMOUDI <ymasmoudi@fb.com>

## Summary

This PR implements helm charts to support eks orchestration of the containerised agw.
This first version uses host network mode until a further solution is found to prevent ovs access from within the pods (mme, sctpd, ...)

The agwc-config pod runs pre-installation/pre-upgrade to build the appropriate configuration for oai-mme, control-proxy and copy the config/rootca.

## Test Plan

Tested with eks in a dedicated cluster.

```
ymasmoudi-mbp:pods ymasmoudi$ kubectl get pods --all-namespaces 
NAMESPACE     NAME                             READY   STATUS    RESTARTS   AGE
kube-system   aws-node-wg442                   1/1     Running   2          32d
kube-system   coredns-54957984bd-88bdk         1/1     Running   1          32d
kube-system   coredns-54957984bd-cql5d         1/1     Running   1          32d
kube-system   kube-proxy-5vwvf                 1/1     Running   1          32d
magma         connectiond-8fd8949f-hm9ll       1/1     Running   0          12m
magma         control-proxy-568f69944b-xrlmr   1/1     Running   0          12m
magma         ctraced-54cc94fdb4-8x7nj         1/1     Running   0          12m
magma         directoryd-64dd9c6bcc-sgvr7      1/1     Running   0          12m
magma         enodebd-65fc659bd5-pxmj9         1/1     Running   0          12m
magma         eventd-598d65859b-kwdh2          1/1     Running   0          12m
magma         health-67594f4574-7gmsk          1/1     Running   0          12m
magma         magmad-7b45fbcf5-gq5n4           1/1     Running   0          12m
magma         mobilityd-677b9b7d6c-v85fc       1/1     Running   2          12m
magma         monitord-54bf887f4b-wq7h4        1/1     Running   0          12m
magma         oai-mme-78c4d54ff4-56gq8         1/1     Running   0          12m
magma         pipelined-cb899c45b-mt64s        1/1     Running   0          12m
magma         policydb-f78f6cc48-bgv7n         1/1     Running   0          12m
magma         redirectd-796899c7c4-fq5lg       1/1     Running   0          12m
magma         redis-7878c64bd9-c9n5p           1/1     Running   0          12m
magma         sctpd-59bb659cf-74jt8            1/1     Running   0          12m
magma         sessiond-85589998f5-7hf2v        1/1     Running   0          12m
magma         smsd-577b6cc8c-ht6xr             1/1     Running   0          12m
magma         state-858c7d66f7-dr9f5           1/1     Running   0          12m
magma         subscriberdb-69dcb856d-wl75k     1/1     Running   0          12m
magma         td-agent-bit-59c8b56cf4-mcqzz    1/1     Running   0          12m
```

### CODE REVIEW

File: lte/gateway/deploy/agwc-helm-charts/templates/agwc-config.pod.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      name: agwc-config
+      resources: {}
+      volumeMounts:
+        - mountPath: /etc/magma
+          name: magma-config
+        - mountPath: /var/opt/magma
+          name: agwc-claim
+        - mountPath: /certs
+          name: agwc-secret
+        - mountPath: /configs
```
Comment (arunuke) [inline lte/gateway/deploy/agwc-helm-charts/templates/agwc-config.pod.yaml:44]: Can the magma-config and controlproxy-config volumes be merged?

File: lte/gateway/deploy/agwc-helm-charts/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# AGW Helm Deployment
+
```
Comment (arunuke) [inline lte/gateway/deploy/agwc-helm-charts/README.md:2]: Would a link to configure the host node with the "base" option only (via the agw docker script) be ideal here? This to ensure the AGW is deployed on a compatible host.

**Review by arunuke — APPROVED** [review]
LGTM !! Documentation and volumes related feedback can be addressed as part of a separate PR if needed.

**Review by koolzz — APPROVED** [review]

**Comment by Alt-Shivam** [comment]
@ymasmoudi 
Is this helm deployment still working fine on your local system?

**Pull request merged**